public/pages/workflow_detail/components/header.tsx

```diff
@@ -52,14 +73,130 @@ export function WorkflowDetailHeader(props: WorkflowDetailHeaderProps) {
     }
   }, [props.workflow]);
 
-  return (
+  const { TopNavMenu, HeaderControl } = getNavigationUI();
+  const { setAppRightControls } = getApplication();
+  const {
+    chrome: { setHeaderVariant },
+  } = getCore();
+  const uiSettings = getUISettings();
+  const showActionsInHeader = uiSettings.get('home:useNewHomePage');
+
```

Comment:
super nit: can you add one-line comments on added `useEffect` hooks to describe what they're doing? especially since this is all pulled logic from core OSD, helps to have some context added if it changes later on.

Reply:
also a little confused on what is happening in here, as it seems to run `setHeaderVariant` twice potentially.